orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,265 @@
+            self.clear()
+            return
+        word_index = self.model().word_index
+        selected_docs = {word_index[index.row()][0] for index in indexes}
+        selected_rows = [row_index for row_index, (doc_index, _) in enumerate(word_index)
+                         if doc_index in selected_docs]
+        selection = QItemSelection()
+        # select all rows belonging to the selected document
+        for row in selected_rows:
+            index = self.model().index(row, 0)
+            selection.select(index, index)
+        super().select(selection, flags)
+
+
+class ConcordanceModel(QAbstractTableModel):
+    "A model for constructing concordances from text."
+
+    def __init__(self):
+        QAbstractTableModel.__init__(self)
+        self.word = None
+        self.data = None
+        self.indices = None
+        self.word_index = None
+        self.width = 8
+
+    def set_word(self, word):
+        self.modelAboutToBeReset.emit()
+        self.word = word
+        if self.indices is None:
+            self.word_index = None
+        else:
+            self.word_index = [(doc_idx, offset) for doc_idx, doc in enumerate(self.indices)
+                               for offset in doc.offsets(self.word)]
+        self.modelReset.emit()
+
+    def set_data(self, data):
+        self.modelAboutToBeReset.emit()
+        self.data = data
+        self._compute_indices()
+        self.modelReset.emit()
+
+    def set_width(self, width):
+        self.modelAboutToBeReset.emit()
+        self.width = width
+        self.modelReset.emit()
+
+    def flags(self, _):
+        return Qt.ItemIsEnabled | Qt.ItemIsSelectable
+
+    def rowCount(self, parent=None, *args, **kwargs):
+        return len(self.word_index) if self.word_index is not None else 0
+
+    def columnCount(self, parent=None, *args, **kwargs):
+        return 3
+
+    def data(self, index, role=Qt.DisplayRole):
+        row, col = index.row(), index.column()
+        doc, index = self.word_index[row]
+
+        if role == Qt.DisplayRole:
+            tokens = self.data.tokens
+            if col == 0:
+                # one more if necessary to assure as much text as possible is returned
+                if index - self.width > 0:
+                    return ' '.join(tokens[doc][index - self.width:index])
+                else:
+                    return ' '.join(tokens[doc][0:index])
+            if col == 1:
+                return tokens[doc][index]
+            if col == 2:
+                if index + self.width < len(tokens[doc]):
+                    return ' '.join(tokens[doc][index + 1:index + self.width])
+                else:
+                    return ' '.join(tokens[doc][index + 1:])
+
+        elif role == Qt.TextAlignmentRole:
+            return [Qt.AlignRight | Qt.AlignVCenter, Qt.AlignCenter, Qt.AlignLeft | Qt.AlignVCenter][col]
+
+        elif role == Qt.ForegroundRole:
+            if col == 1:
+                return QBrush(QColor(0, 0, 255))
+
```

Comment:
Well, obviously. 😁  That's what I was trying to do. My approach:
```
    def set_width(self):
        sel = self.conc_view.selectionModel().selection()
        self.model.set_width(self.width)
        self.conc_view.selectionModel().select(sel)
```

`select()`, however, requires me to pass flags as well and I don't know how to do this and why aren't they stored in sel. 😞